includes/wikia/services/UserService.class.php

```diff
@@ -86,28 +97,60 @@ public function resetPassword( User $targetUser ) {
 		return $tempPass;
 	}
 
+	/**
+	 * Generates password reset token and sends it to user via email
+	 *
+	 * @param User   $targetUser
+	 * @param string $returnUrl    Url the user will be redirected to after setting new password
+	 * @param string $tokenContext Used to choose the correct email template, so far only "facebook" value causes to use
+	 *                             a dedicated facebook-disconnect email
+	 *
+	 * @return boolean true for success, false otherwise
+	 */
+	public function requestResetToken( User $targetUser, $returnUrl, $tokenContext ) {
+		/** @var HeliosClient $heliosClient */
+		$heliosClient = Injector::getInjector()->get( HeliosClient::class );
+		$result = $heliosClient->requestPasswordReset( $targetUser->getId(), $returnUrl, $tokenContext );
+
+		if ( !empty( $result->success ) ) {
+			return true;
+		}
+
+		$log = WikiaLogger::instance();
+		$log->error( 'Failed to request a password reset token', [
+			'tokenContext' => $tokenContext,
+			'user_id'      => $targetUser->getId(),
+			'error'        => empty( $result->errors ) ? 'unknown-error' : $result->errors[0]->description,
+
+		] );
+
+		return false;
+	}
+
 	/** Helper methods for getUsers */
 
 	/**
 	 * Methods builds User object depending on Ids and Names in ids array
+	 *
 	 * @param $ids array list of user ids and names to look for
+	 *
 	 * @return array with User objects
 	 */
 	private function getUsersObjects( $ids ) {
 		wfProfileIn( __METHOD__ );
-		$result = array();
+		$result = [];
 
-		if( isset( $ids[ 'user_id' ] ) ) {
-			foreach( $ids[ 'user_id' ] as $id ) {
+		if ( isset( $ids['user_id'] ) ) {
+			foreach ( $ids['user_id'] as $id ) {
 				$user = User::newFromId( $id );
 				//skip default user
 				if ( $user && $user->getTouched() != 0 ) {
 					$result[] = $user;
 				}
 			}
 		}
-		if( isset( $ids[ 'user_name' ] ) ) {
-			foreach( $ids[ 'user_name' ] as $name ) {
+		if ( isset( $ids['user_name'] ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "user_name" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)